src/contracts/core/RewardsCoordinator.sol

```diff
@@ -178,41 +172,56 @@ contract RewardsCoordinator is
+        PerformanceRewardsSubmission[] calldata performanceRewardsSubmissions
+    ) external onlyWhenNotPaused(PAUSED_AVS_PERFORMANCE_REWARDS_SUBMISSION) nonReentrant {
+        for (uint256 i = 0; i < performanceRewardsSubmissions.length; i++) {
+            PerformanceRewardsSubmission calldata performanceRewardsSubmission = performanceRewardsSubmissions[i];
+            uint256 nonce = submissionNonce[msg.sender];
+            bytes32 performanceRewardsSubmissionHash = keccak256(
+                abi.encode(msg.sender, nonce, performanceRewardsSubmission)
+            );
+
+            uint256 totalAmount = _validatePerformanceRewardsSubmission(performanceRewardsSubmission);
+
+            isAVSPerformanceRewardsSubmissionHash[msg.sender][performanceRewardsSubmissionHash] = true;
+            submissionNonce[msg.sender] = nonce + 1;
+
+            emit AVSPerformanceRewardsSubmissionCreated(
+                msg.sender,
+                nonce,
+                performanceRewardsSubmissionHash,
+                performanceRewardsSubmission
+            );
+            performanceRewardsSubmission.token.safeTransferFrom(msg.sender, address(this), totalAmount);
+        }
+    }
+
     /// @inheritdoc IRewardsCoordinator
     function processClaim(
         RewardsMerkleClaim calldata claim,
         address recipient
     ) external onlyWhenNotPaused(PAUSED_PROCESS_CLAIM) nonReentrant {
-        DistributionRoot memory root = _distributionRoots[claim.rootIndex];
-        _checkClaim(claim, root);
-        // If claimerFor earner is not set, claimer is by default the earner. Else set to claimerFor
-        address earner = claim.earnerLeaf.earner;
-        address claimer = claimerFor[earner];
-        if (claimer == address(0)) {
-            claimer = earner;
-        }
-        require(msg.sender == claimer, "RewardsCoordinator.processClaim: caller is not valid claimer");
-        for (uint256 i = 0; i < claim.tokenIndices.length; i++) {
-            TokenTreeMerkleLeaf calldata tokenLeaf = claim.tokenLeaves[i];
-
-            uint256 currCumulativeClaimed = cumulativeClaimed[earner][tokenLeaf.token];
-            require(
-                tokenLeaf.cumulativeEarnings > currCumulativeClaimed,
-                "RewardsCoordinator.processClaim: cumulativeEarnings must be gt than cumulativeClaimed"
-            );
-
-            // Calculate amount to claim and update cumulativeClaimed
-            uint256 claimAmount = tokenLeaf.cumulativeEarnings - currCumulativeClaimed;
-            cumulativeClaimed[earner][tokenLeaf.token] = tokenLeaf.cumulativeEarnings;
+        _processClaim(claim, recipient);
+    }
 
-            tokenLeaf.token.safeTransfer(recipient, claimAmount);
-            emit RewardsClaimed(root.root, earner, claimer, recipient, tokenLeaf.token, claimAmount);
+    /// @inheritdoc IRewardsCoordinator
+    function processClaims(
+        RewardsMerkleClaim[] calldata claims,
+        address recipient
```

Comment:
We could also claim for multiple recipients instead of claiming to a single recipient. Thoughts?

Reply:
single recipient seems fine